src/scanspec/specs.py

```diff
@@ -600,7 +614,98 @@ def bounded(
 """
 Defers wrapping function with validate_call until class is fully instantiated
 """
-Line.bounded = validate_call(Line.bounded)  # type:ignore
+Linspace.bounded = validate_call(Linspace.bounded)  # type:ignore
+
+
+@dataclass(config=StrictConfig)
+class Range(Spec[Axis]):
+    """Linearly spaced frames with start and stop as first and last midpoints.
+
+    This class is intended to handle linearly spaced frames defined with a
+    specific step size.
+
+    .. seealso::
+        `Linspace`: For linearly spaced frames defined with a number of frames.
+
+    .. example_spec::
+
+        from scanspec.specs import Fly, Range
+
+        spec = Fly(Range("x", 1, 2, 0.25))
+    """
+
+    axis: Axis = Field(description="An identifier for what to move")
+    start: float = Field(description="Midpoint of the first point of the line")
+    stop: float = Field(description="Midpoint of the last point of the line")
+    step: float | None = Field(
+        description="Step size (defaults to stop - start)", default=None
+    )
+
+    def __post_init__(self):
+        if self.step == 0:
+            raise ValueError("step must be nonzero")
+        if self.step is None:
+            self.step = self.stop - self.start
+
+    def axes(self) -> list[Axis]:  # noqa: D102
+        return [self.axis]
+
+    def _line_from_indexes(
+        self, indexes: npt.NDArray[np.float64]
+    ) -> dict[Axis, npt.NDArray[np.float64]]:
+        step = abs(self.step) * np.sign(self.stop - self.start)  # type: ignore
```

Comment:
...then you can get rid of the type ignores:

```suggestion
        step = abs(self.step) * np.sign(self.stop - self.start)
```